kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/dek/Dek.java

```diff
@@ -221,7 +115,7 @@ private static long commenceDestroyBoth(long combined) {
             throw new IllegalArgumentException();
         }
         if (remainingEncryptions.addAndGet(-numEncryptions) >= 0) {
-            if (encryptorCount(outstandingCryptors.updateAndGet(Dek::acquireEncryptor)) <= 0) {
+            if (!outstandingCryptors.acquireEncryptorUsage()) {
```

Comment:
This is the win for me. It makes it much clearer whats gong on in this method and why.

Reply:
AFAICS there's nothing stopping us from writing this method in the existing `Dek` though. I guess we didn't so far because each of these new methods is only getting called once, but if your argument is that the code is clearer then we can do that. 

And if you want the extra test coverage make those methods package-private and static, accepting an `AtomicLong` and write a separate test just for the reference counting.